allow services to determine which fields to persist via samson/persis…

grosser opened the pull request:
…tent_fields

@ragurney @jonmoter @jasonmartens

jonmoter commented on plugins/kubernetes/app/models/kubernetes/resource.rb:
```diff
@@ -210,7 +210,8 @@ def template_for_update
           "spec.clusterIP",
-          *ENV["KUBERNETES_SERVICE_PERSISTENT_FIELDS"].to_s.split(",")
+          *ENV["KUBERNETES_SERVICE_PERSISTENT_FIELDS"].to_s.split(","),
+          *copy.dig(:metadata, :annotations, :"samson/persistent_fields").to_s.split(",")
```
Maybe `split(',').map(&:strip)`?

grosser commented:
supporting space and comma now

On Fri, Feb 23, 2018 at 11:24 AM, Jon Moter <notifications@github.com>
wrote:

> *@jonmoter* commented on this pull request.
> ------------------------------
>
> In plugins/kubernetes/app/models/kubernetes/resource.rb
> <https://github.com/zendesk/samson/pull/2602#discussion_r170343715>:
>
> > @@ -210,7 +210,8 @@ def template_for_update
>          [
>            "metadata.resourceVersion",
>            "spec.clusterIP",
> -          *ENV["KUBERNETES_SERVICE_PERSISTENT_FIELDS"].to_s.split(",")
> +          *ENV["KUBERNETES_SERVICE_PERSISTENT_FIELDS"].to_s.split(","),
> +          *copy.dig(:metadata, :annotations, :"samson/persistent_fields").to_s.split(",")
>
> Maybe split(',').map(&:strip)?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/2602#pullrequestreview-99023976>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ4IrEHf6fZ1KHZa0HtYVwEYpOJbjks5tXxB0gaJpZM4SRWu2>
> .
>
